### PR TITLE
feat(balance): fix stacking error with heavy bones, updates to harvest entries and recipes, use new materials in mods

### PR DESCRIPTION
**Pull request by chaosvolt**
### Required

- [X] I wrote the PR title in [conventional commit format](https://docs.cataclysmbn.org/en/contribute/changelog_guidelines/).
- [X] I ran the [code formatter](https://docs.cataclysmbn.org/en/contribute/contributing/#code-style).
- [X] I linked any relevant issues using [github keyword syntax](https://docs.cataclysmbn.org/en/contribute/contributing/#pull-request-notes) so it can be closed automatically.
- [X] I have [committed my changes to new branch that isn't `main`](https://docs.cataclysmbn.org/en/contribute/contributing/#make-your-changes) so it won't cause conflict when updating `main` branch later.

## Purpose of change

A collection of assorted followups to the addition of monster body materials.

## Describe the solution

1. Fixed "spawned with charges but can't have any" issues with heavy bones and the like via setting heavy bones, teeth, and scutes to be stackable. Also set basic looks_like entries for all of them.
2. Adjusted harvest entries to make it so only skeletal brutes and juggernauts get heavy bones, with regular skeletal zombies having tainted bones as normal.
3. Added a harvest entry to allow mutant carp and salmon to have scutes and heavy bones, while sewer fish get scutes but stick to normal bones.
4. Rigged harvest entry for evolved mutants to allow them to have pointy teeth and heavy bones.
5. Fixed boneless flesh golems and jabberwocks, they now get to be big-boned.
6. Fixed it so mutant bears and coyotes get to have pointy teeth, and mutant bears specifically get heavy bones. Scaled bears also reflavored to be scaly instead of chitinous.
7. Allowed zombie versions of large animals to also have heavy bones, and for all zombie versions of carnivorus animals to have teeth.
8. Set it so tough zombies no longer have heavy bones, resembling them instead for brute and hulk-tier monsters. Nightstalkers, shocker brutes/husks, and kevlar zeds get heavy bones as a result.
9. Fixed weapons made from heavy bone and serrated teeth not actually using the material defined for them.
10. Set stingers and stinger daggers to be made out of chitin.
11. Added heavy bones as an alternative to bones in some recipes.
12. Updated relevant creatures in mods. Zombie uplifts in Aftershock get bones/teeth where relevant, dinomod dinos also updated, etc.
13. Misc: Removed MORE GODDAMN redundant overrides in No Hope blyat.

## Describe alternatives you've considered

screm

## Testing

Checked affected files for syntax and lint errors.

### CODE REVIEW

**Comment by chaosvolt** [comment]
Just spotted one more mistake to fix: sharktooth arrows don't have recipes.

File: data/json/monsters/mammal.json [hunk 1]
```diff
@@ -808,7 +808,7 @@
     "melee_cut": 0,
     "dodge": 3,
     "vision_night": 12,
-    "harvest": "mammal_large_leather",
+    "harvest": "mutant_mammal_leather",
```
Comment (NappingOcean) [inline data/json/monsters/mammal.json:811]: Are deers mutated? I guess it's a mistake.

Comment (chaosvolt) [reply data/json/monsters/mammal.json:811]: Whoops, thanks for spotting that.

**Review by RoyalFox2140 — APPROVED** [review]

**Comment by Proxiehunter** [comment]
Is the error this fixes when this build goes out the probable reason I just did a quick butchery on a mutant rattle snake with a survival skill of 4 and a hunting knife and got nothing or should I report the error?

**Comment by chaosvolt** [comment]
AAAAAAAAAAAAAA

**Comment by chaosvolt** [comment]
Wait. WAIT.

I didn't even touch rattlesnakes...?

**Pull request merged**